tests/test_storage.py

```diff
@@ -2,17 +2,25 @@
 from unittest.mock import ANY
 
 import pytest
-from bleak.backends.device import BLEDevice
 from bleak.backends.scanner import AdvertisementData
 
 from habluetooth.storage import (
+    BLEDeviceDict,
     DiscoveredDeviceAdvertisementData,
     DiscoveredDeviceAdvertisementDataDict,
     discovered_device_advertisement_data_from_dict,
     discovered_device_advertisement_data_to_dict,
     expire_stale_scanner_discovered_device_advertisement_data,
 )
 
+from . import generate_ble_device
```

Comment:
We should revert all this and add tests that we can still restore BLEDevice when it has rssi and ignore it